File: 42-trapping-rain-water/trapping-rain-water.java
class Solution {
    public int trap(int[] height) {
        int sum=0,n=height.length;
        for(int i=0;i<n;i++){
            int m1=0,m2=0;
            for(int j=0;j<=i;j++){
                m1=Math.max(m1,height[j]);
            }
            for(int j=n-1;j>=i;j--){
                m2=Math.max(m2,height[j]);
            }
            sum=sum+(Math.min(m2,m1)-height[i]);
        }
        return sum;
    }
}

File: 42-trapping-rain-water/trapping-rain-water.py
class Solution(object):
    def trap(self, height):
        l=height[:]
        r=height[:]
        ml=-99999 
        sum=0
        mr=-99999
        for i in range(0,len(height)):
            ml=max(ml,height[i])
            l[i]=ml

        for i in range(len(height)-1,-1,-1):
            mr=max(mr,height[i])
            r[i]=mr
        for i in range(0,len(height)):
            sum=sum+(min(r[i],l[i])-height[i])
        return sum
                